doc/source/walkthrough.rst

```diff
@@ -215,131 +381,278 @@ Object IDs can be created in multiple ways.
+      }
 
-.. autofunction:: ray.wait
-    :noindex:
+      try {
+        rayObject.get();
+      } catch (RayTaskException e) {
+        Throwable rootCause = e.getCause();
+        Assert.assertTrue(rootCause instanceof RuntimeException);
+      }
+
+    After launching a number of tasks, you may want to know which ones have
+    finished executing. This can be done with ``Ray.wait``. The function
+    works as follows.
+
+    .. code:: java
+
+        WaitResult<Integer> waitResult = Ray.wait(rayObjectList, /*num_returns=*/1, /*timeoutMs=*/1000);
+        System.out.println(waitResult.getReady());  // List of ready objects.
+        System.out.println(waitResult.getUnready());  // list of unready objects.
 
 Object Eviction
 ---------------
 
-When the object store gets full, objects will be evicted to make room for new objects.
-This happens in approximate LRU (least recently used) order. To avoid objects from
-being evicted, you can call ``ray.get`` and store their values instead. Numpy array
-objects cannot be evicted while they are mapped in any Python process. You can also
-configure `memory limits <memory-management.html>`__ to control object store usage by
-actors.
+.. tabs::
 
-.. note::
+  .. group-tab:: Python
+
+    When the object store gets full, objects will be evicted to make room for new objects.
+    This happens in approximate LRU (least recently used) order. To avoid objects from
+    being evicted, you can call ``ray.get`` and store their values instead. Numpy array
+    objects cannot be evicted while they are mapped in any Python process. You can also
+    configure `memory limits <memory-management.html>`__ to control object store usage by
+    actors.
+
+    .. note::
 
-    Objects created with ``ray.put`` are pinned in memory while a Python reference
-    to the object ID returned by the put exists. This only applies to the specific
-    ID returned by put, not IDs in general or copies of that IDs.
+        Objects created with ``ray.put`` are pinned in memory while a Python reference
+        to the object ID returned by the put exists. This only applies to the specific
+        ID returned by put, not IDs in general or copies of that IDs.
+
+  .. group-tab:: Java
+
+    When the object store gets full, objects will be evicted to make room for new objects.
+    This happens in approximate LRU (least recently used) order. To avoid objects from
+    being evicted, you can call ``RayObject::get`` or ``Ray.get`` and store their values instead.
+    You can also configure `memory limits <memory-management.html>`__ to control object store usage by
+    actors.
+
+    .. note::
+
+        Objects created with ``Ray.put`` are pinned in memory while a Java reference
+        to the ``RayObject`` returned by the put exists. This only applies to the specific
+        ``RayObject``-s returned by put, not ``RayObject``-s in general or copies of thoses ``RayObject``-s.
```

Comment:
Pinning in Java is not implemented yet. We need to rewrite this section for Java.

Reply:
Since it seems like there are going to be some limitations/significant differences for Ray on Java across the board, I recommend that we put all the java information in its own page. In doing so we make it extremely simple to reference for users using Java / wanting to get started and we get a single place that we can call everything out in terms of advantages or limitations.

This also makes sense since Tune / RLlib / SGD / etc. aren't going to be supported.

Also, I'm going to do a large scale overall of the core docs to make it easier to search and organize. If you like, I am more than happy to include the Java docs information in my PR.

Reply:
I thought about this a bit more and think this is all a great addition! Please ignore my previous comment and let's get this merged for the next release!